### PR TITLE
KFP V2 upgrades

### PR DESCRIPTION
**Pull request by ghost**
# Description

* Updated repo for KFP V2
* Updated Python version to 3.9
* Updated training step to use `@container_component` with a custom training container
* Implemented a custom serving container
* Simplified champion/challenger deployment
* Removed Tensorflow pipelines to simplify codebase
* Compiled pipelines now uploaded to Artifact Registry instead of GCS
* CI/CD and developer scripts updated in line with the above
* Lots of documentation updates!

# How has this been tested?

* Successfully run training + prediction pipelines end-to-end (both with and without a champion-challenger comparison)
* Tested developer scripts manually
* Tested CI/CD pipelines (incl release pipeline to compile+publish pipelines)

# Checklist

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have successfully run the E2E tests, and have included the links to the pipeline runs below
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have updated any relevant documentation to reflect my changes
- [x] I have assigned a reviewer and messaged them

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -105,3 +105,9 @@ destroy-infra: ## DESTROY the Terraform infrastructure in your project. Requires
 	@ cd terraform/envs/$(env) && \
 	terraform init -backend-config='bucket=${VERTEX_PROJECT_ID}-tfstate' && \
 	terraform destroy -var 'project_id=${VERTEX_PROJECT_ID}' -var 'region=${VERTEX_LOCATION}'
+
+build-training-container: ## Build and push training container image using Docker
+	@ cd training && \
+	poetry export -f requirements.txt -o requirements.txt && \
+	docker build -t ${TRAINING_CONTAINER_IMAGE} . && \
+	docker push ${TRAINING_CONTAINER_IMAGE}
```
Comment (felix-datatonic) [inline Makefile]: can we use `gcloud builds` instead?

File: components/vertex-components/src/vertex_components/upload_model.py [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+
+@component(
+    base_image="python:3.9",
+    packages_to_install=[
+        "google-cloud-aiplatform==1.28.1",
+        "google-cloud-pipeline-components==2.1.0",
+    ],
+)
+def upload_model(
```
Comment (felix-datatonic) [inline components/vertex-components/src/vertex_components/upload_model.py:26]: good idea to wrap this into a single component

**Comment by ghost** [comment]
TODO - fix poetry installation in CI/CD

**Review by felix-datatonic — APPROVED** [review]

**Pull request merged**